### DISCUSSION TITLE
[Realm SYNC] Do I need a partner collection for a non-destructive change?.

### DISCUSSION THREAD

**Discussion by vardansargsyan92** [realm/realm-dotnet#2529] [Q&A] [answered]
Hello. 

Imagine that we have a `Person` object in my realm database:

```
public class Person : RealmObject
{
    [PrimaryKey]
    [MapTo("_id")]
    public ObjectId Id { get; set; }

    public string FirstName { get; set; }
    public string LastName { get; set; }
    public int Age { get; set; }
}
```

Then I decide to use combined `FullName` property instead of the separate `FirstName` and `LastName` property in the Person schema.

```
public class Person : RealmObject
{
    [PrimaryKey]
    [MapTo("_id")]
    public ObjectId Id { get; set; }

    public string FullName { get; set; }
    public int Age { get; set; }
}
```
 According to the [doc](https://docs.mongodb.com/realm/sync/modify-synced-object-schema/) this change isn't destructive and I don't have to create a partner collection, forward and backward migration functions. 
 
 But the problem is that I would like to migrate the realm to conform to the updated `Person` schema. If I were developing an application without Realm Sync, I would use the following migration function:
 
 
```
 var config = new RealmConfiguration
{
    SchemaVersion = 4,
    MigrationCallback = (migration, oldSchemaVersion) =>
    {
        var oldPeople = migration.OldRealm.All<Person>();
        var newPeople = migration.NewRealm.All<Person>();

        // Migrate Person objects
        for (var i = 0; i < newPeople.Count(); i++)
        {
            var oldPerson = oldPeople.ElementAt(i);
            var newPerson = newPeople.ElementAt(i);

            // Changes from version 1 to 2 (adding LastName) will occur automatically when Realm detects the change

            // Migrate Person from version 2 to 3: replace FirstName and LastName with FullName
            // LastName doesn't exist in version 1
            if (oldSchemaVersion < 2)
            {
                newPerson.FullName = oldPerson.FirstName;
            }
            else if (oldSchemaVersion < 3)
            {
                newPerson.FullName = $"{oldPerson.FirstName} {oldPerson.LastName}";
            }
        }
    }
};
var realm = Realm.GetInstance(config);
```

and that would be enough to extract the `FullName` from older databases combining the `FirstName` and `LastName` properties.

So I have the following questions:

1. Do I need a partner collection for that changes?
2. Does the non-sync realm migration function applicable for the realm sync? And if No what I should do in that case?

**Comment by ianpward** [comment, marked as answer]
hey @vardansargsyan92 can you move this question to forums.realm.io please and ping me? I can get someone to respond. In general the github repo's are for non-sync related bugs - for design and implementation questions related to sync please use the forums.

**Reply by vardansargsyan92** [reply]
hi @ianpward. Unfortunately I am not able to move it to forums because I am not able to sign in. When I try to log in to system it's says _Sorry, you can't access Community Forums (New) because you are not assigned this app in Okta. If you're wondering why this is happening, please contact your administrator._ 

![image](https://user-images.githubusercontent.com/19687305/126277530-bd43216a-93e7-4f7a-b08a-3b4857dc92c8.png)

**Reply by vardansargsyan92** [reply]
@ianpward seems the issue has been fixed. So here is a [question](https://www.mongodb.com/community/forums/t/realm-sync-do-i-need-a-partner-collection-for-a-non-destructive-change/115814)
